content/blog/2017/2017-01-12-declarative-pipeline-beta-2.adoc

```diff
@@ -0,0 +1,122 @@
+---
+layout: post
+title: "Declarative Pipeline Syntax Beta 2 release"
+tags:
+- plugins
+- pipeline
+author: abayer
+---
+
+This week, we released the second beta of the new
+link:https://wiki.jenkins-ci.org/display/JENKINS/Pipeline+Model+Definition+Plugin[Declarative Pipeline syntax],
+available in the Update Center now as version 0.8.1 of _Pipeline: Model Definition_.
+You can read more about Declarative Pipeline
+link:/blog/2016/12/19/declarative-pipeline-beta/[in the blog post introducing the first beta]
+from December, but we wanted to update you all on the syntax changes in the
+second beta. These syntax changes are the last compatibility-breaking changes to
+the syntax for the 1.0 release planned for February, so you can safely start
+using the 0.8.1 syntax now without needing to change it when 1.0 is released.
+
+== Syntax Changes
+
+=== Agent
+In order to support more detailed and clear configuration of agents, as well as
+making agent syntax more consistent with the rest of the Declarative Pipeline
+syntax, we've moved the agent configuration into blocks. The `agent any` and
+`agent none` configurations work the same as previously, but `label`, `docker`
+and `dockerfile` now look like the following:
+
+[pipeline]
+----
+// Just specifying a label is simple.
+agent {
+    label "some-label"
+}
+
+// If you're just specifying a Docker image, you can use this simple syntax.
+agent {
+    docker "ubuntu:lts"
+}
+
+// When you are specifying a label or other arguments, docker looks like this:
+agent {
+    docker {
+        image "ubuntu:lts"
+        label "docker-label"
+        args "-v /tmp:/tmp -p 8000:8000"
+    }
+}
+
+// When you're building an image from "Dockerfile" in your repository and
+// don't care what node is used or have additional arguments, you can again
+// use a simple syntax.
+agent {
+    dockerfile true
+}
+
+// When you're building an image from a differnt file, or have a label or other
+// arguments, use the following syntax:
+agent {
+    dockerfile {
+        filename "OtherDockerfile"
+        label "docker-label"
+        args "-v /tmp:/tmp -p 8000:8000"
+    }
+}
+----
```

Comment:
These various `agent` syntaxes all in one `[pipeline]` block I think is confusing. Ideally (IMO) they should be separated into their own blocks with the comment above each agent declaration as the paragraph of text outside the block